Lib/test/test_urlparse.py

```diff
@@ -936,6 +936,16 @@ def test_issue14072(self):
         self.assertEqual(p2.scheme, 'tel')
         self.assertEqual(p2.path, '+31641044153')
 
+    def test_issue33034(self):
+        # Test case to asset a valid port as an integer
+        p1 = urllib.parse.urlparse('https://www.python.org:80')
+        self.assertEqual(p1.port, 80)
+
+        # Test case to assert ValueError when a string is parsed as a port
+        p2 = urllib.parse.urlparse('http://Server=sde; Service=sde:oracle')
+        with self.assertRaises(ValueError):
```

Comment:
Since the exception type is unchanged and only the exception message is changed, I think we also need to add an assert for the new exception message.

Reply:
Great idea.  Update to the PR coming right up.